merge_checks.py

```diff
@@ -2,16 +2,16 @@
 import subprocess
 
 
-def main(base_ref: str, head_ref: str):
-    print(f"Checking out BASE_REF ({base_ref}) and HEAD_REF ({head_ref})...")
-
+def main(head_hash: str, base_ref: str = "master"):
+    print(f"Checking out {base_ref}...")
     subprocess.run(f"git fetch --depth=1 origin {base_ref}", check=True, shell=True)
-    subprocess.run(f"git fetch origin {head_ref}", check=True, shell=True)
-
     base_revision = subprocess.run(f"git rev-parse origin/{base_ref}", check=True, shell=True, capture_output=True)
-    head_revision = subprocess.run(f"git rev-parse origin/{head_ref}", check=True, shell=True, capture_output=True)
 
-    base_hash, head_hash = (output.decode().strip() for output in (base_revision.stdout, head_revision.stdout))
+    base_hash = base_revision.stdout.decode().strip()
+
+    if head_hash == base_hash:
+        print(f"HEAD identical with {base_ref}, no commits to check!")
+        exit(0)
```

Comment:
Kriege ich die Pipeline grün, indem ich den HEAD einfach nochmal force-pushe?

Reply:
Ich bin mir nicht ganz sicher ob ich dich hier richtig verstehe. Kannst du den Vorgang an den du hier denkst genauer beschreiben?

Reply:
Ich pushe auf den Feature-Branch, sehe die Pipeline ist rot, und mache dann einfach ein `push -f`, also nur ein Push-Event ohne tatsächliche Änderungen, d.h. `head_hash == base_hash`. In dem Fall kommt ein `exit(0)` raus, also wird das wieder grün, richtig?

Reply:
Ah, danke für die Erklärung. Das sollte keine Probleme machen. `base_hash` ist von master, `head_hash` vom Feature-Branch, damit sollte der von dir beschriebene Fall nicht auftreten können.

Ich habe das auch zusätzlich noch einmal ausprobiert. GitHub scheint bei einen (force) push ohne Änderungen keine neue Push-Events auszulösen, das heißt selbst die Pipeline würde auch nicht erneut laufen.